### ISSUE TITLE
Mishaps aren't working on most (if not all) Hexcuteif patterns

### ISSUE THREAD

**Issue by LevAlTru** [N08I40K/hexecuteif#2]
### The problem:
The patterns of Hexcuteif do not produce mishaps in places where needed.
Hexcuteif does the mishap logic but produces an error in console
The error message is at the bottom ↓

### How to reproduce:
1) Play on Forge 1.19.2
2) Write this:
![image](https://github.com/N08I40K/hexecuteif/assets/68346194/a4441e7e-c138-4535-9667-caed8430efd2)
2.1) Have the first slot of your hotbar empty
3) Write
![image](https://github.com/N08I40K/hexecuteif/assets/68346194/44d6919b-697d-42ab-b7bc-b469c209c3c5)
4) Perceive a gray pattern and an error in your Minecraft console
![image](https://github.com/N08I40K/hexecuteif/assets/68346194/c4305921-b16b-4486-916e-5621ef3ee527)

### Additional information:
The error message:
```
[20:44:50] [Server thread/ERROR] [minecraft/BlockableEventLoop]: Error executing task on Server
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 3
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?] {}
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?] {}
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[?:?] {}
	at java.util.Objects.checkIndex(Objects.java:359) ~[?:?] {re:mixin}
	at java.util.ArrayList.set(ArrayList.java:441) ~[?:?] {re:mixin}
	at at.petrak.hexcasting.api.spell.mishaps.MishapInvalidIota.execute(MishapInvalidIota.kt:25) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:classloading}
	at at.petrak.hexcasting.api.spell.casting.sideeffects.OperatorSideEffect$DoMishap.performEffect(OperatorSideEffect.kt:95) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading}
	at at.petrak.hexcasting.api.spell.casting.CastingHarness.performSideEffects(CastingHarness.kt:356) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading,pl:mixin:A}
	at at.petrak.hexcasting.api.spell.casting.CastingHarness.executeIotas(CastingHarness.kt:140) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading,pl:mixin:A}
	at at.petrak.hexcasting.api.spell.casting.CastingHarness.handler$bhk000$executeIotaMacro(CastingHarness.kt:1019) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading,pl:mixin:A}
	at at.petrak.hexcasting.api.spell.casting.CastingHarness.executeIota(CastingHarness.kt) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading,pl:mixin:A}
	at at.petrak.hexcasting.common.network.MsgNewSpellPatternSyn.lambda$handle$0(MsgNewSpellPatternSyn.java:92) ~[hexcasting-forge-1.19.2-0.10.3.jar%23374!/:0.10.3] {re:mixin,re:classloading,pl:mixin:APP:hexgloop-common.mixins.json:ring.MixinRingSendPatterns,pl:mixin:A}
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.essential.json:client.Mixin_ThreadTaskExecutor,pl:mixin:A}
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:computing_frames,re:mixin,re:classloading}
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:763) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:157) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.essential.json:client.Mixin_ThreadTaskExecutor,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:746) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:740) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:116) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:mixins.essential.json:client.Mixin_ThreadTaskExecutor,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:725) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:658) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[client-1.19.2-20220805.130853-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,xf:fml:xaerominimap:xaero_minecraftserver,pl:mixin:APP:create_new_age.mixins.json:MinecraftServerMixin,pl:mixin:APP:fastload.mixins.json:server.MinecraftServerMixin,pl:mixin:APP:crafttweaker.mixins.json:common.access.server.AccessMinecraftServer,pl:mixin:APP:valkyrienskies-common.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixins.essential.json:feature.sps.Mixin_IntegratedServerResourcePack,pl:mixin:APP:mixins.essential.json:server.MinecraftServerMixin_PvPGameRule,pl:mixin:APP:mixins.essential.json:server.Mixin_PublishServerStatusResponse,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:833) [?:?] {re:mixin}
```

**Comment by n08i40k** [comment]
Will be fixed in the next release.

**Comment by n08i40k** [comment]
Fixed in release 0.0.5!
